mypy/semanal_namedtuple.py

```diff
@@ -196,17 +232,27 @@ def parse_namedtuple_args(self, call: CallExpr,
                 items = [cast(StrExpr, item).value for item in listexpr.items]
             else:
                 # The fields argument contains (name, type) tuples.
-                items, types, ok = self.parse_namedtuple_fields_with_types(listexpr.items, call)
+                items, types, _, ok = self.parse_namedtuple_fields_with_types(listexpr.items, call)
         if not types:
             types = [AnyType(TypeOfAny.unannotated) for _ in items]
         underscore = [item for item in items if item.startswith('_')]
         if underscore:
             self.fail("namedtuple() field names cannot start with an underscore: "
                       + ', '.join(underscore), call)
-        return items, types, ok
+        if num_defaults > len(items):
+            self.fail("Too many defaults given in call to namedtuple()", call)
+            num_defaults = len(items)
+        return items, types, num_defaults, ok
+
+    def extract_defaults_from_arg(self, call: CallExpr, arg: Expression) -> int:
+        if not isinstance(arg, (ListExpr, TupleExpr)):
```

Comment:
Technically the defaults argument can be any iterable, but it doesn't seem particularly useful to support that.